### PR TITLE
Bootstrap RTL the correct way

### PR DESCRIPTION
**Pull request by brianteeman**
This PR is using bootstrap the way that bootstrap intended for it to be used since 5.0 for RTL.

removing any rtl specific code that can be automatically handled by rtlcss or by using rtlcss specific markup (just as bootstrap itself does)

All our current css is doing bootstrap the wrong way for rtl support.

The problem is NOT with our own scss as in general this is now correct with logical properties (The few instances of dir=rtl are now handled by rtlcss)

The problem is with the actual bootstrap code. We are using it without addressing any RTL issues with that. With the exception of a very incomplete set of bootstrap overrides in https://github.com/joomla/joomla-cms/blob/7cf5e1228db337881d7678ad1c71915d5c885150/build/media_source/templates/administrator/atum/scss/vendor/bootstrap/_bootstrap-rtl.scss

Technically this will change all left to right etc so we dont need to use logical properties in our own css however I still think its good practice that we continue to use logical properties where available.

Bootstrap 5 is designed to be post-processed using rtlcss - This takes the generated template.css and converts it to template-rtl.css

## Benefits
- it is DRY
- we don't need to remember to check it works in rtl (we almost always forget)
- we dont need to waste time creating rtl overrides (in most cases)
- fixes numerous open issues
- replaces numerous open pull requests
- fixes several issues with input buttons not already on the tracker.

## Testing
1. Apply the pr and then run `npm i`
This will generate all the css and install rtlcss

Or

2.Use a prebuuilt package

3. test test test

### Before
image

### After
image

### Before
image

### After
image

### CODE REVIEW

**Comment by dgrammatiko** [comment]
> With this PR we need to update the build tools. rtlcsss is a postcss plugin that runs on the generated template.css of atum and cassiopeia and after generating the new -rtl.css this needs to be minified as well.

https://github.com/brianteeman/joomla-cms/pull/291

**Comment by brianteeman** [comment]
Thanks @dgrammatiko

**Review by wilsonge — APPROVED** [review]
AMAZING if this works. I looked when I plumbed in 5.x and it seemed really painful to migrate everything. Thought it was way more work than this. Really nice job!

**Comment by brianteeman** [comment]
> Thought it was way more work than this. 

Well it was 14hours solid so not a small amount - plus dont forget all the PR since you did the first work where I moved our own css to logical properties

**Comment by wilsonge** [comment]
Yeah that makes a lot of sense. Still really nice work though :)

**Comment by brianteeman** [comment]
Thanks

File: build/media_source/templates/site/cassiopeia/scss/blocks/_footer.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@
   margin-top: $cassiopeia-grid-gutter;
   color: $white;
   background-color: var(--cassiopeia-color-primary);
-  background-image: $cassiopeia-header-grad;
+  background-image: $cassiopeia-header-grad  #{"/* rtl:"}$cassiopeia-header-grad-rtl #{"*/"};
```
Comment (Quy) [inline build/media_source/templates/site/cassiopeia/scss/blocks/_footer.scss]:
```suggestion
  background-image: $cassiopeia-header-grad #{"/* rtl:"}$cassiopeia-header-grad-rtl #{"*/"};
```

**Comment by brianteeman** [comment]
is there any point in me fixing the conflicts here or is it just going to be ignored like so many pull requests recently?

**Comment by chmst** [comment]
Your PR's are not ignored .. there are only not enough active testers.

**Comment by laoneo** [comment]
This should be rebased to 4.3 anyway.

**Comment by brianteeman** [comment]
its a bug fix not a feature

**Comment by laoneo** [comment]
19 changed files and a new package is a too big change for a bug fix release. I welcome the change, but we really need to bring 4.2 into a stable state, so stuff like this needs to go into the next minor. I know we both have different opinions what a bug fix is, but we should start to handle patch releases as such ones and not getting the risk to become unstable again, even when the chance is small. I really hope you understand that point, this here is not a risky one. Again, the change is very welcome but please for 4.3.

**Comment by brianteeman** [comment]
rebased as requested - of ciurse its now untestable until 4.3 brnch is brought up to date

**Comment by laoneo** [comment]
Thanks, merging stuff like this as early as possible brings us more time to iron out issues, even when the chance is small.

**Comment by brianteeman** [comment]
except it cant be tested right now :(

> Thanks, merging stuff like this as early as possible brings us more time to iron out regressions, even when the chance is small that there will be one.

I hope this means that pull requests wont wait as RTC for weeks and months before merging any more

**Comment by laoneo** [comment]
Updated the branch, should be back to normal.

**Comment by brianteeman** [comment]
Thanks @laoneo

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 4609b33164939f64dbe574aeade304388d129735<br /><br />Tested it with farsi. Did browse around and saw no issue. Installed DPCalendar and created events and so on. All did look ok with RTL.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38412">issues.joomla.org/tracker/joomla-cms/38412</a>.</sub>

**Comment by brianteeman** [comment]
thanks for testing. dpc was one of the components I developed this with ;)

File: build/media_source/templates/site/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss [hunk 2]
```diff
@@ -7,7 +7,7 @@
 .form-select {
   max-width: $input-max-width;
   cursor: pointer;
-  background: $form-select-background;
+  background: $form-select-background #{"/* rtl:"}$form-select-background-rtl#{"*/"};;
```
Comment (Quy) [inline build/media_source/templates/site/cassiopeia/scss/vendor/bootstrap/_custom-forms.scss]:
```suggestion
  background: $form-select-background #{"/* rtl:"}$form-select-background-rtl#{"*/"};
```

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 95ba902e072a87d0aeed8c3f1acf21ad1f8e657b<br /><br />I did extensive browsing trying to catch issues. 
As a non-rtl reader, I may have missed issues related to rtl that I may be unaware of, but I honestly think this PR was a great work through.
It would be useful to get a tester using rtl daily.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38412">issues.joomla.org/tracker/joomla-cms/38412</a>.</sub>

**Comment by alikon** [comment]
made some tests, but i'm not a RTL user...
...so if we don't get any RTL tester... you can consider this as a valid one

**Comment by brianteeman** [comment]
fyi I use RTL a lot which is why I spent the time making the pr

**Comment by obuisard** [comment]
I would like to merge this PR into 4.3 early on. I am adding RTC status to get people's attention, but I will merge it tomorrow if I don't get any objection by then.

**Comment by brianteeman** [comment]
is it tomorrow yet?

**Comment by obuisard** [comment]
Was hoping to hear from reviewers...

**Comment by obuisard** [comment]
Thank you Brian @brianteeman for this needed fix and for keeping track.

**Pull request merged**